packages/maskbook/src/plugins/RedPacket/services.ts

```diff
@@ -73,3 +75,19 @@ export async function getRedPacketsFromChain(from: string, startBlock: number) {
     if (response.status !== 200) return []
     return response.json() as Promise<History.RedPacketRecord[]>
 }
+
+export async function getAllRedPackets(address: string) {
+    const chainId = await getChainId()
+    const redPacketsFromChain = await subgraph.getAllRedPackets(address)
+    const redPacketsFromDB = await database.getRedPacketsHistory(redPacketsFromChain.map((x) => x.rpid))
+    return redPacketsFromChain.reduce((acc, history) => {
+        const record = redPacketsFromDB.find((y) => y.payload.rpid === history.rpid)
+        if (history.chain_id === chainId && record) {
```

Comment:
Check `history.chain_id === chainId` first if it's true the finding logic can be ommited.

Reply:
It can't be ommited since we need password.